nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -1250,6 +1255,54 @@ export class FlowEffects {
         { dispatch: false }
     );
 
+    openEditRemoteProcessGroupDialog$ = createEffect(
+        () =>
+            this.actions$.pipe(
+                ofType(FlowActions.openEditRemoteProcessGroupDialog),
+                map((action) => action.request),
+                tap((request) => {
+                    const editDialogReference = this.dialog.open(EditRemoteProcessGroup, {
+                        data: request,
+                        panelClass: 'large-dialog'
+                    });
+
+                    editDialogReference.componentInstance.saving$ = this.store.select(selectSaving);
+
+                    editDialogReference.componentInstance.editRemoteProcessGroup
+                        .pipe(takeUntil(editDialogReference.afterClosed()))
+                        .subscribe((payload: any) => {
+                            this.store.dispatch(
+                                FlowActions.updateComponent({
+                                    request: {
+                                        id: request.entity.id,
+                                        uri: request.uri,
+                                        type: request.type,
+                                        payload
+                                    }
+                                })
+                            );
+                        });
+
+                    editDialogReference.afterClosed().subscribe(() => {
+                        this.store.dispatch(FlowActions.clearFlowApiError());
+                        this.store.dispatch(
+                            FlowActions.selectComponents({
+                                request: {
+                                    components: [
+                                        {
+                                            id: request.entity.id,
+                                            componentType: request.type
+                                        }
+                                    ]
+                                }
+                            })
+                        );
```

Comment:
We should conditionally dispatch this action like we do in other `EditXyz` dialogs if the user isn't routed elsewhere. We have an interceptor that may forward to an error page. If that happens we have a fallback to ensure that all dialogs are closed and the close payload will be `ROUTED`. If this were to occur we wouldn't want `selectComponents` to be dispatched.